core/vm/operations_verkle.go

```diff
@@ -104,15 +108,15 @@ func gasSelfdestructEIP4762(evm *EVM, contract *Contract, stack *Stack, mem *Mem
 	contractAddr := contract.Address()
 	statelessGas := evm.AccessEvents.VersionGas(contractAddr, false)
 	statelessGas += evm.AccessEvents.CodeSizeGas(contractAddr, false)
-	statelessGas += evm.AccessEvents.BalanceGas(contractAddr, false)
+	statelessGas += evm.AccessEvents.BalanceGas(contractAddr, false, false)
 	if contractAddr != beneficiaryAddr {
-		statelessGas += evm.AccessEvents.BalanceGas(beneficiaryAddr, false)
+		statelessGas += evm.AccessEvents.BalanceGas(beneficiaryAddr, false, false)
 	}
 	// Charge write costs if it transfers value
 	if evm.StateDB.GetBalance(contractAddr).Sign() != 0 {
-		statelessGas += evm.AccessEvents.BalanceGas(contractAddr, true)
+		statelessGas += evm.AccessEvents.BalanceGas(contractAddr, true, false)
 		if contractAddr != beneficiaryAddr {
-			statelessGas += evm.AccessEvents.BalanceGas(beneficiaryAddr, true)
+			statelessGas += evm.AccessEvents.BalanceGas(beneficiaryAddr, true, !evm.StateDB.Exist(beneficiaryAddr))
```

Comment:
same point my @jsign : the account will need to be created so the fill might be more than just the balance

Reply:
this one should be fixed by just charging the empty code hash as well from here